multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/TimeoutAsyncFlowableStep.java

```diff
@@ -50,4 +54,35 @@ private String getStepName() {
     }
 
     public abstract Duration getTimeout(ProcessContext context);
+
+    protected Integer extractUploadTimeoutFromAppAttributes(CloudApplicationExtended app, String supportedParameter) {
+        ApplicationAttributes appAttributes = ApplicationAttributes.fromApplication(app, app.getEnv());
+        Number taskExecutionTimeout = appAttributes.get(supportedParameter, Number.class);
+        return taskExecutionTimeout != null ? taskExecutionTimeout.intValue() : null;
+    }
+
+    protected void logTimeout(String message, Number timeout) {
+        getStepLogger().debug(message, timeout);
+    }
+
+    protected Duration getGivenTimeout(ProcessContext context, String supportedParameter, Variable<Duration> durationOperational,
+                                       Variable<Duration> durationGlobal, String message) {
+        CloudApplicationExtended app = context.getVariable(Variables.APP_TO_PROCESS);
+        Integer timeout = extractUploadTimeoutFromAppAttributes(app, supportedParameter);
+        Duration timeoutOperational = context.getVariable(durationOperational);
+
+        Duration resultTimeout;
+        if (timeoutOperational != null) {
+            resultTimeout = timeoutOperational;
+        } else if (timeout != null) {
+            resultTimeout = Duration.ofSeconds(timeout);
+        } else {
+            int timeoutGlobal = (int) context.getVariable(durationGlobal)
+                                             .toSeconds();
+            resultTimeout = Duration.ofSeconds(timeoutGlobal);
+        }
+
+        logTimeout(message, resultTimeout.toSeconds());
```

Comment:
make message the same but parametrized - additionally can log all levels of the variable and make clear which one is the most important/resolved as